engine/src/main/java/org/terasology/rendering/dag/nodes/CopyImageToHMDNode.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.nodes;
+
+import jopenvr.JOpenVRLibrary;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.rendering.dag.ConditionDependentNode;
+import org.terasology.rendering.dag.stateChanges.EnableMaterial;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FBOConfig;
+import org.terasology.rendering.openvrprovider.OpenVRProvider;
+import org.lwjgl.opengl.GL11;
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.registry.In;
+import static org.lwjgl.opengl.GL11.*;
+import org.terasology.rendering.opengl.ScreenGrabber;
+import org.terasology.rendering.opengl.fbms.DisplayResolutionDependentFBOs;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.rendering.world.WorldRenderer.RenderingStage;
+
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.FINAL;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+import static org.terasology.rendering.opengl.ScalingFactors.FULL_SCALE;
+
+public class CopyImageToHMDNode extends ConditionDependentNode {
+    private static final ResourceUrn LEFT_EYE_FBO = new ResourceUrn("engine:leftEye");
+    private static final ResourceUrn RIGHT_EYE_FBO = new ResourceUrn("engine:rightEye");
+    // TODO: make these configurable options
+
+    @In
+    private OpenVRProvider vrProvider;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    @In
+    private ScreenGrabber screenGrabber;
+
+    @In
+    private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
+
+    private RenderingConfig renderingConfig;
+    private FBO leftEye;
+    private FBO rightEye;
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        requiresCondition(() -> (renderingConfig.isVrSupport()
+                && vrProvider != null
+                && vrProvider.isInitialized() ));
+        leftEye = requiresFBO(new FBOConfig(LEFT_EYE_FBO, FULL_SCALE,
+                FBO.Type.DEFAULT).useDepthBuffer(),displayResolutionDependentFBOs);
+        rightEye = requiresFBO(new FBOConfig(RIGHT_EYE_FBO,FULL_SCALE,
+                FBO.Type.DEFAULT).useDepthBuffer(),displayResolutionDependentFBOs);
+        if (this.vrProvider != null) {
```

Comment:
vrProvider can only be null if isVrSupport() turns out false, I guess. These checks ended up in here because, although the name would suggest that we'll never end up hitting this code in certain conditions, the system was a bit unfamiliar to me so I wrote some extra checks. I can respect the desire for brevity so I'll remove them in the upcoming commit.